packages/core/__tests__/middleware.js

```diff
@@ -76,47 +80,35 @@ test('middleware options', () => {
 
   expect(actual).toBe(expected)
 })
-test("middleware's custom emitter", () => {
+
+test('enhance channel.put with an emitter', () => {
   const actual = []
+  const channel = stdChannel()
+  const rawPut = channel.put
+  channel.put = action => {
+    if (action.type === 'batch') {
+      action.batch.forEach(rawPut)
+      return
+    }
+    rawPut(action)
+  }
 
   function* saga() {
     yield takeEvery('*', ac => actual.push(ac.type))
```

Comment:
regarding my comment here - https://github.com/redux-saga/redux-saga/pull/1654/files#r228758007 

```js
yield takeEvery('*', function*(ac) {
  actual.push(ac.type)
  yield put({ type: 'saga_action' })
})
```

because of making `dispatch` "optional" you have made this put here not hitting reducers automatically